packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -49,6 +49,15 @@ export type CustomSelectButtonProps = {
 	value?: string | string[];
 };
 
+// Props only exposed on the internal implementation
+export type _CustomSelectInternalProps = {
+	/**
+	 * True if the consumer is emulating the legacy component behavior and look
+	 */
+	isLegacy?: boolean;
+};
```

Comment:
I needed a way to allow the internal implementation to expose a `isLegacy` flag, without forwarding it to the public API of the V2 component. Not sure if we think there's a better way to do that? @mirka

Reply:
Can't think of a better way, TBH. 

What a bummer that we have to do it with hacks like this. It's an indication that the legacy adapter approach has probably too many downsides and won't be a recommended one in the future.

Reply:
Yeah it may not be worth to keep a common underlying implementation for V1 and V2 if we then have to abstract it and complicate it with extra props and exceptions. Let's keep it as-is for now and look back at it once V2 legacy wrapper has shipped and we look at V2